packages/core/status/core-status-server-internal/src/log_plugins_status.ts

```diff
@@ -0,0 +1,187 @@
+      ({ isReportedStatus }) => isReportedStatus
+    );
+
+    const inferredCount = pluginStatuses.length - statusReportingPlugins.length;
+
+    if (statusReportingPlugins.length === 1) {
+      const reason = getReason(statusReportingPlugins[0]);
+      const allAvailableSuffix =
+        status.available.length === total ? ` (all ${total} plugins are now available)` : '';
+
+      if (inferredCount === 0) {
+        messages.push(
+          `'${statusReportingPlugins[0].pluginName}' is now ${currentLevel}: ${reason}${allAvailableSuffix}`
+        );
+      } else {
+        messages.push(
+          `'${statusReportingPlugins[0].pluginName}' (and ${inferredCount} more) are now ${currentLevel}: ${reason}${allAvailableSuffix}`
+        );
+      }
+    } else if (statusReportingPlugins.length > 1) {
+      const multilineStatus = [];
+      if (inferredCount === 0) {
+        multilineStatus.push(
+          `${
+            statusReportingPlugins.length
+          } plugins are now ${currentLevel}: ${statusReportingPlugins
+            .map(({ pluginName }) => pluginName)
+            .join(', ')}`
+        );
+      } else {
+        multilineStatus.push(
+          `The following plugins are now ${currentLevel}: ${statusReportingPlugins
+            .map(({ pluginName }) => pluginName)
+            .join(', ')} (and ${inferredCount} more)`
+        );
+      }
+
+      statusReportingPlugins.forEach(({ pluginName, ...pluginStatus }) =>
+        multilineStatus.push(
+          ` - '${pluginName}' is now ${currentLevel}: ${getReason(pluginStatus)}`
+        )
+      );
+
+      messages.push(multilineStatus.join('\n'));
+    }
+  });
+
+  // report plugins that are STILL unhealthy (even though their status has not changed)
+  Object.entries(status)
+    .filter(([currentLevel]) => currentLevel !== ServiceStatusLevels.available.toString())
+    .forEach(([currentLevel, pluginStatuses]) => {
+      const statusReportingPlugins = pluginStatuses
+        .filter(
+          (pluginStatus) =>
+            !updates[currentLevel as PluginLevel].find(
+              (updatedStatus) => updatedStatus.pluginName === pluginStatus.pluginName
+            )
+        )
+        .filter(({ isReportedStatus }) => isReportedStatus);
+
+      const inferredCount = pluginStatuses.length - statusReportingPlugins.length;
+
+      if (statusReportingPlugins.length === 1) {
+        const reason = getReason(statusReportingPlugins[0]);
+        if (inferredCount === 0) {
+          messages.push(`${statusReportingPlugins[0].pluginName} is ${currentLevel}: ${reason}`);
+        } else {
+          messages.push(
+            `'${statusReportingPlugins[0].pluginName}' plugin (and some others that depend on it) are ${currentLevel}: ${reason}`
+          );
+        }
+      } else if (statusReportingPlugins.length > 1) {
+        if (inferredCount === 0) {
+          messages.push(
+            `The following plugins are ${currentLevel}: ${statusReportingPlugins.map(
```

Comment:
NIT: message inconsistency between
- `statusReportingPlugins.length === 1` => `{pluginName} is ${currentLevel}`
- `statusReportingPlugins.length > 1` => `The following plugins are ${currentLevel}`.

I think more consistency would make sense, e.g `The following plugin...` and `The following plugins...`

Reply:
When there is a single plugin changing to a specific status, I am putting the `${reason}` in the same line, thus the distinction.

I will try to unify and preserve consistency.

Reply:
It's a NIT, honestly, but I'm mostly thinking about us here, and I think it would help us for log discovery (e.g searching for "the following" or things like that.

Reply:
Mixing this comment and the `some` ones... from o11y POV, we want to know which plugin(s) caused the unavailability. From a user perspective, I would expect something like:
* `Kibana is now unavailable (was available). [#] plugins are now unavailable. Status change caused by plugin(s) ${listOfPluginsThatCausedTheDegradedState}.`

As an extra, I find it useful that we are printing the reason for the plugins turning unavailable (maybe it can be printed via the logger `status.${plugin}`?) Although, based on recent experience, we may risk logging information that shouldn't be logged 🙈, so I don't know anymore 🤷.

If it's OK to log the reason, I wonder if the status could be placed as part of the meta of the "general" log so, one log entry would have: 
1. The message: as defined above: summary of the global status, the number of plugins unavailable + which plugins caused the status change
2. The meta: it contains the status at that point in time

WDYT?

Reply:
Sounds like a desirable log, although there are a few challenges to achieve this.
Kibana status change can be caused by degraded plugins but also by degraded core services (saved objects, elasticsearch client). We will need a way to find out which of them caused the status change.